File: flutter_app_1771020345/lib/models/menu_item.dart
class MenuItem {
  final int id;
  final String name;
  final double price;
  final String description;
  final String imageUrl;
  final bool isVegetarian;
  final bool isSpicy;
  final int preparationTime;

  MenuItem({
    required this.id,
    required this.name,
    required this.price,
    required this.description,
    required this.imageUrl,
    required this.isVegetarian,
    required this.isSpicy,
    required this.preparationTime,
  });

  factory MenuItem.fromJson(Map<String, dynamic> json) {
    return MenuItem(
      id: json['id'],
      name: json['name'],
      price: (json['price'] as num).toDouble(),
      description: json['description'] ?? '',
      imageUrl: json['image_url'] ?? '',
      isVegetarian: json['is_vegetarian'] ?? false,
      isSpicy: json['is_spicy'] ?? false,
      preparationTime: json['preparation_time'] ?? 0,
    );
  }
}

File: backend/web_api_1771020345/Models/MenuItem.cs
using System.ComponentModel.DataAnnotations;
namespace web_api_1771020345.Models;

public class MenuItem
{
    [Key]
    public int Id { get; set; }
    [Required]
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    [Required]
    public string Category { get; set; } = null!;
    [Required]
    public decimal Price { get; set; }
    public string? ImageUrl { get; set; }
    public int PreparationTime { get; set; }
    public bool IsVegetarian { get; set; } = false;
    public bool IsSpicy { get; set; } = false;
    public bool IsAvailable { get; set; } = true;
    public decimal Rating { get; set; } = 0;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}
